### DISCUSSION TITLE
Attempting to Integrate

### DISCUSSION THREAD

**Discussion by MichaelBMiner** [rauc/meta-rauc#255] [Q&A]
Hello,

I am following your README and I have added rauc to my image. When I go to build I get the error 
`ERROR: Layer rauc is not compatible with the core layer which only supports these series: hardknott (layer is compatible with langdale)
`

I am using Yocto from NXP and my distro is hardknott. Can anyone explain what is happening here?

**Comment by ejoerns** [comment]
Hi @MichaelBMiner,

this is a standard mechanism of Yocto to explicitly maintain the compatibility of different layers.
It is more or less a statement that the layer has been developed for a specific Yocto(/poky) release. The release name is 'hardknott' in your case. For a more detailed explanation see:

https://docs.yoctoproject.org/ref-manual/variables.html#term-LAYERSERIES_COMPAT

When having a 'hardknott' BSP (which is EOL since April 2021 btw!), standard practice is to use the corresponding 'hardknott' branch of all other layers (if available). Thus simply git checkout the [hardknott](https://github.com/rauc/meta-rauc/tree/hardknott) branch of meta-rauc and the error should disappear.

**Reply by MichaelBMiner** [reply]
That did work, it gives me a new issue that only appears after adding DISTRO_FEATURES
`Please ensure that your setting of VIRTUAL-RUNTIME_init_manager (systemd) matches the entries enabled in DISTRO_FEATURES`

**Reply by ejoerns** [reply]
The manual only suggests to enable `rauc` in `DISTRO_FEATURES`. I don't see how this changes init manager.

You can use `bitbake <recipename> -e | less` to dump and inspect the environment for how `DISTRO_FEATURES` are set.

**Reply by MichaelBMiner** [reply]
Getting closer. I have a customised image so I added the following to my local.conf
```
DISTRO_FEATURES += "rauc"
CORE_IMAGE_EXTRA_INSTALL += "rauc"
INIT_MANAGER = "systemd"
```

It cannot seem to find my rauc version. I have added the recipe rauc in recipes core. I have rauc_%.bbappends added as well.
```
rauc/rauc_git.bb: Unable to get checksum for rauc SRC_URI entry post-install.sh: file could not be found
rauc/rauc_1.6.bb: Unable to get checksum for rauc SRC_URI entry post-install.sh: file could not be found
```

**Comment by MichaelBMiner** [comment]
Thank you for the support. I think I almost have it. My image fails on the rootfs step and gives a long output 

```
ERROR: evccapplication-dev-1.0-r0 do_rootfs: Unable to install packages. Command '/mnt/WorkDrive/Documents/EVCC_Application/build-fb/tmp/work/imx6ull14x14evk-poky-linux-gnueabi/evccapplication-dev/1.0-r0/recipe-sysroot-native/usr/bin/opkg --volatile-cache -f /mnt/WorkDrive/Documents/EVCC_Application/build-fb/tmp/work/imx6ull14x14evk-poky-linux-gnueabi/evccapplication-dev/1.0-r0/opkg.conf -t /mnt/WorkDrive/Documents/EVCC_Application/build-fb/tmp/work/imx6ull14x14evk-poky-linux-gnueabi/evccapplication-dev/1.0-r0/temp/ipktemp/ -o /mnt/WorkDrive/Documents/EVCC_Application/build-fb/tmp/work/imx6ull14x14evk-poky-linux-gnueabi/evccapplication-dev/1.0-r0/rootfs  --force_postinstall --prefer-arch-to-version   --add-ignore-recommends udev-hwdb install base-passwd boost bridge-utils can-utils crccalculator crda dbcppp debugconfigs dhcpcd dtc ethtool evcccan evccconfigs iperf3 isose iw kernel-module-lwb5p-backports-laird kernel-modules laird-localize-firmware libgpiod libsocketcan logrotate lumissil lumissil-kernel-module lwb5plus-sdio-sa-firmware mainapplication open-plc-utils openssl packagegroup-core-boot packagegroup-core-eclipse-debug packagegroup-core-ssh-openssh rauc run-postinsts shadow socat stress-ng stressapptest sudo systemd-analyze tcpdump udevrules wireless-regdb-static wireless-tools wpa-supplicant' returned 255:
Disfavor package: udev-hwdb
Installing base-passwd (3.5.29) on root

```
